src/js/api.js

```diff
@@ -1,14 +1,50 @@
-import {generateApiUrl} from './utils';
-import {renderMovie} from './templates';
+import {generateApiUrl, generateSearchUrl} from './utils';
+import {renderMovies, renderError, renderDetails, renderSearch} from './templates';
 
 export const getMovies = (path, selector) => {
-  fetch(generateApiUrl(path))
-    .then(res => res.json())
-    .then(({results}) => {
-      console.log('results', results)
+    fetch(generateApiUrl(path))
+        .then(res => res.json())
+        .then(data => {
+            console.log('results', data.results);
 
-      const movies = results.map(movie => renderMovie(movie)).join('');
+            const root = document.querySelector(selector);
 
-      document.querySelector(selector).insertAdjacentHTML('beforeend', movies);
-    });
+            if(data.status_message && data.status_code) {
+                root.innerHTML = renderError({ status_code: data.status_code, status_message: data.status_message });
+            } else {
+                root.innerHTML = renderMovies(data.results);
+            }
+        });
 }
+
+export const getMovieDetails = (path, selector) => {
+    fetch(generateApiUrl(path))
+        .then(res => res.json())
+        .then(data => {
+            console.log('data', data);
+
+            const root = document.querySelector(selector);
+
+            if(data.status_message && data.status_code) {
+                root.innerHTML = renderError({ status_code: data.status_code, status_message: data.status_message });
+            } else {
+                root.innerHTML = renderDetails(data);
+            }
+        });
+}
+
+export const SearchMovie = (path, selector) => {
```

Comment:
Across all project we use camel case for naming, so will be better if you named this like -`searchMovie` or `getSearchMovie`